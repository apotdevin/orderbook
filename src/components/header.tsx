import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  height: 40px;
  background: rgba(0, 0, 0, 0.8);
  width: 100%;
  margin-bottom: 32px;
`;

export const Header = () => {
  return <Wrapper>Orderbook Example</Wrapper>;
};
