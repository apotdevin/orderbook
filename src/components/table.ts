import styled, { css } from 'styled-components';
import { Colors } from '../styles/Colors';

export const TableStyles = styled.div`
  overflow-x: auto;
  table {
    border-spacing: 0;

    thead {
      color: white;
    }

    tr {
      :last-child {
        td {
          border-bottom: 0;
        }
      }
    }

    th,
    td {
      font-size: 14px;
      text-align: right;
      margin: 0;
      padding: 2px 0;
      width: 100px;
      height: 20px;
      :last-child {
        border-right: 0;
      }
    }
  }
`;

export const TableRow = styled.tr<{
  inverted?: boolean;
  maxDepth: number;
  currentValue: number;
  left?: boolean;
}>`
  background-image: ${({ inverted, maxDepth, currentValue, left }) => {
    const percent = 100 - Math.round((currentValue / maxDepth) * 100);

    return inverted
      ? css`linear-gradient(to left, transparent ${percent}%, ${Colors.lightRed} ${percent}%);`
      : css`linear-gradient(to ${
          left ? 'left' : 'right'
        }, transparent ${percent}%, ${Colors.lightGreen} ${percent}%);`;
  }};
`;
