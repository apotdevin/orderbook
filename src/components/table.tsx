import styled, { css } from 'styled-components';
import { useTable, useSortBy, useGlobalFilter } from 'react-table';
import { Colors } from 'src/styles/Colors';

type StyledTableProps = {
  withBorder?: boolean;
  alignCenter?: boolean;
  fontSize?: string;
};

const Styles = styled.div`
  overflow-x: auto;
  table {
    border-spacing: 0;

    thead {
      color: black;
    }

    tr {
      :last-child {
        td {
          border-bottom: 0;
        }
      }
    }

    th,
    td {
      font-size: ${({ fontSize }: StyledTableProps) => fontSize || '14px'};
      text-align: right;
      margin: 0;
      padding: 2px 0;
      width: 100px;
      height: 20px;
      ${({ withBorder }: StyledTableProps) =>
        withBorder &&
        css`
          border-bottom: 1px solid black;
        `}
      ${({ alignCenter }: StyledTableProps) =>
        alignCenter &&
        css`
          text-align: center;
          padding: 8px;
        `}
      :last-child {
        border-right: 0;
      }
    }
  }
`;

const Row = styled.tr<{
  inverted?: boolean;
  maxDepth: number;
  currentValue: number;
}>`
  background-image: ${({ inverted, maxDepth, currentValue }) => {
    const percent = 100 - Math.round((currentValue / maxDepth) * 100);

    return inverted
      ? css`linear-gradient(to left, transparent ${percent}%, ${Colors.lightRed} ${percent}%);`
      : css`linear-gradient(to right, transparent ${percent}%, ${Colors.lightGreen} ${percent}%);`;
  }};
`;

type TableProps = {
  inverted?: boolean;
  tableData: any[];
  tableColumns:
    | { Header: string; accessor: string }[]
    | { Header: string; columns: { Header: string; accessor: string }[] }[];
  withBorder?: boolean;
  fontSize?: string;
  filterPlaceholder?: string;
  notSortable?: boolean;
};

export const Table: React.FC<TableProps> = ({
  inverted,
  tableData,
  tableColumns,
  withBorder,
  fontSize,
}) => {
  const data = tableData;
  const columns = tableColumns;

  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    rows,
    prepareRow,
  } = useTable(
    {
      columns,
      data,
    },
    useGlobalFilter,
    useSortBy
  );

  return (
    <Styles withBorder={withBorder} fontSize={fontSize}>
      <table {...getTableProps()} style={{ width: '100%' }}>
        <thead>
          {headerGroups.map((headerGroup, index) => (
            <tr {...headerGroup.getHeaderGroupProps()} key={index}>
              {headerGroup.headers.map((column, index) => (
                <th key={index}>{column.render('Header')}</th>
              ))}
            </tr>
          ))}
        </thead>
        <tbody {...getTableBodyProps()}>
          {rows.map((row, index) => {
            prepareRow(row);
            return (
              <Row
                {...row.getRowProps()}
                key={`${(row.original as { price: number }).price}-${index}`}
                inverted={inverted}
                maxDepth={(row.original as { max: number }).max || 1}
                currentValue={(row.original as { depth: number }).depth || 0}
              >
                {row.cells.map((cell, index) => {
                  return (
                    <td {...cell.getCellProps()} key={index}>
                      {cell.render('Cell')}
                    </td>
                  );
                })}
              </Row>
            );
          })}
        </tbody>
      </table>
    </Styles>
  );
};
