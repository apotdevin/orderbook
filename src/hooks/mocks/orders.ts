import { OrderEntry } from '../useOrderbook';

export const OrderMocks: OrderEntry[] = [
  {
    price: 1,
    size: 100,
    depth: 0,
  },
  {
    price: 2,
    size: 120,
    depth: 0,
  },
  {
    price: 10,
    size: 1000,
    depth: 0,
  },
  {
    price: 30,
    size: 7,
    depth: 0,
  },
];

export const IncomingOrderMocks: [number, number][] = [
  [1, 2],
  [30, 12],
];

export const ResultMock: OrderEntry[] = [
  {
    depth: 12,
    price: 30,
    size: 12,
  },
  {
    depth: 1012,
    price: 10,
    size: 1000,
  },
  {
    depth: 1132,
    price: 2,
    size: 120,
  },
  {
    depth: 1134,
    price: 1,
    size: 2,
  },
];

export const ResultNoIncomingMock: OrderEntry[] = [
  {
    depth: 7,
    price: 30,
    size: 7,
  },
  {
    depth: 1007,
    price: 10,
    size: 1000,
  },
  {
    depth: 1127,
    price: 2,
    size: 120,
  },
  {
    depth: 1227,
    price: 1,
    size: 100,
  },
];

export const ResultAskMock: OrderEntry[] = [
  {
    depth: 2,
    price: 1,
    size: 2,
  },
  {
    depth: 122,
    price: 2,
    size: 120,
  },
  {
    depth: 1122,
    price: 10,
    size: 1000,
  },
  {
    depth: 1134,
    price: 30,
    size: 12,
  },
];

export const IncomingOrderWithEmptyMocks: [number, number][] = [
  [1, 240],
  [2, 0],
  [10, 0],
  [30, 1200],
];

export const ResultWithEmptySizesMock: OrderEntry[] = [
  {
    depth: 240,
    price: 1,
    size: 240,
  },
  {
    depth: 1440,
    price: 30,
    size: 1200,
  },
];

export const IncomingOrderMocksForGrouping: [number, number][] = [
  [100.5, 130],
  [101, 260],
  [101.5, 120],
  [102, 250],
  [125, 120],
  [127, 123],
  [130, 332],
  [170, 2230],
  [132, 93],
  [151, 455],
  [199, 305],
  [210, 203],
  [270, 733],
  [285, 1733],
  [290, 379],
];

export const GroupedByTwoFiveMock: OrderEntry[] = [
  {
    depth: 379,
    price: 290,
    size: 379,
  },
  {
    depth: 2112,
    price: 285,
    size: 1733,
  },
  {
    depth: 2845,
    price: 270,
    size: 733,
  },
  {
    depth: 3048,
    price: 210,
    size: 203,
  },
  {
    depth: 3353,
    price: 197.5,
    size: 305,
  },
  {
    depth: 5583,
    price: 170,
    size: 2230,
  },
  {
    depth: 6038,
    price: 150,
    size: 455,
  },
  {
    depth: 6463,
    price: 130,
    size: 425,
  },
  {
    depth: 6706,
    price: 125,
    size: 243,
  },
  {
    depth: 7466,
    price: 100,
    size: 760,
  },
];

export const GroupedByFiveMock: OrderEntry[] = [
  {
    depth: 379,
    price: 290,
    size: 379,
  },
  {
    depth: 2112,
    price: 285,
    size: 1733,
  },
  {
    depth: 2845,
    price: 270,
    size: 733,
  },
  {
    depth: 3048,
    price: 210,
    size: 203,
  },
  {
    depth: 3353,
    price: 195,
    size: 305,
  },
  {
    depth: 5583,
    price: 170,
    size: 2230,
  },
  {
    depth: 6038,
    price: 150,
    size: 455,
  },
  {
    depth: 6463,
    price: 130,
    size: 425,
  },
  {
    depth: 6706,
    price: 125,
    size: 243,
  },
  {
    depth: 7466,
    price: 100,
    size: 760,
  },
];

export const GroupedByTenMock: OrderEntry[] = [
  {
    depth: 379,
    price: 290,
    size: 379,
  },
  {
    depth: 2112,
    price: 280,
    size: 1733,
  },
  {
    depth: 2845,
    price: 270,
    size: 733,
  },
  {
    depth: 3048,
    price: 210,
    size: 203,
  },
  {
    depth: 3353,
    price: 190,
    size: 305,
  },
  {
    depth: 5583,
    price: 170,
    size: 2230,
  },
  {
    depth: 6038,
    price: 150,
    size: 455,
  },
  {
    depth: 6463,
    price: 130,
    size: 425,
  },
  {
    depth: 6706,
    price: 120,
    size: 243,
  },
  {
    depth: 7466,
    price: 100,
    size: 760,
  },
];

export const GroupedByFiftyMock: OrderEntry[] = [
  {
    depth: 2845,
    price: 250,
    size: 2845,
  },
  {
    depth: 3048,
    price: 200,
    size: 203,
  },
  {
    depth: 6038,
    price: 150,
    size: 2990,
  },
  {
    depth: 7466,
    price: 100,
    size: 1428,
  },
];

export const GroupedByOneHundredMock: OrderEntry[] = [
  {
    depth: 3048,
    price: 200,
    size: 3048,
  },
  {
    depth: 7466,
    price: 100,
    size: 4418,
  },
];

export const GroupedByTwoFiveAsksMock: OrderEntry[] = [
  {
    depth: 760,
    price: 102.5,
    size: 760,
  },
  {
    depth: 880,
    price: 125,
    size: 120,
  },
  {
    depth: 1003,
    price: 127.5,
    size: 123,
  },
  {
    depth: 1335,
    price: 130,
    size: 332,
  },
  {
    depth: 1428,
    price: 132.5,
    size: 93,
  },
  {
    depth: 1883,
    price: 152.5,
    size: 455,
  },
  {
    depth: 4113,
    price: 170,
    size: 2230,
  },
  {
    depth: 4418,
    price: 200,
    size: 305,
  },
  {
    depth: 4621,
    price: 210,
    size: 203,
  },
  {
    depth: 5354,
    price: 270,
    size: 733,
  },
  {
    depth: 7087,
    price: 285,
    size: 1733,
  },
  {
    depth: 7466,
    price: 290,
    size: 379,
  },
];

export const GroupedByFiveAsksMock: OrderEntry[] = [
  {
    depth: 760,
    price: 105,
    size: 760,
  },
  {
    depth: 880,
    price: 125,
    size: 120,
  },
  {
    depth: 1335,
    price: 130,
    size: 455,
  },
  {
    depth: 1428,
    price: 135,
    size: 93,
  },
  {
    depth: 1883,
    price: 155,
    size: 455,
  },
  {
    depth: 4113,
    price: 170,
    size: 2230,
  },
  {
    depth: 4418,
    price: 200,
    size: 305,
  },
  {
    depth: 4621,
    price: 210,
    size: 203,
  },
  {
    depth: 5354,
    price: 270,
    size: 733,
  },
  {
    depth: 7087,
    price: 285,
    size: 1733,
  },
  {
    depth: 7466,
    price: 290,
    size: 379,
  },
];

export const GroupedByTenAsksMock: OrderEntry[] = [
  {
    depth: 760,
    price: 110,
    size: 760,
  },
  {
    depth: 1335,
    price: 130,
    size: 575,
  },
  {
    depth: 1428,
    price: 140,
    size: 93,
  },
  {
    depth: 1883,
    price: 160,
    size: 455,
  },
  {
    depth: 4113,
    price: 170,
    size: 2230,
  },
  {
    depth: 4418,
    price: 200,
    size: 305,
  },
  {
    depth: 4621,
    price: 210,
    size: 203,
  },
  {
    depth: 5354,
    price: 270,
    size: 733,
  },
  {
    depth: 7466,
    price: 290,
    size: 2112,
  },
];

export const GroupedByFiftyAsksMock: OrderEntry[] = [
  {
    depth: 1428,
    price: 150,
    size: 1428,
  },
  {
    depth: 4418,
    price: 200,
    size: 2990,
  },
  {
    depth: 4621,
    price: 250,
    size: 203,
  },
  {
    depth: 7466,
    price: 300,
    size: 2845,
  },
];

export const GroupedByOneHundredAsksMock: OrderEntry[] = [
  {
    depth: 4418,
    price: 200,
    size: 4418,
  },
  {
    depth: 7466,
    price: 300,
    size: 3048,
  },
];
