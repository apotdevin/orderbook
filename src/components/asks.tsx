import { FC } from 'react';
import { OrderEntry } from 'src/hooks/useOrderbook';
import { TableRow, TableStyles } from 'src/components/table';
import { formatCurrency } from 'src/utils/format';
import { Typo } from './typo';

export const Asks: FC<{ entries: OrderEntry[]; max: number }> = ({
  entries,
  max,
}) => {
  const tableData = entries.map(b => ({
    ...b,
    depthStr: formatCurrency(b.depth, 0),
    sizeStr: formatCurrency(b.size, 0),
    priceStr: <Typo.ask>{formatCurrency(b.price)}</Typo.ask>,
    max,
  }));

  return (
    <TableStyles>
      <table>
        <thead>
          <tr>
            <td>PRICE</td>
            <td>SIZE</td>
            <td>TOTAL</td>
          </tr>
        </thead>
        <tbody>
          {tableData.map((entry, index) => (
            <TableRow
              key={`${entry.price}-${index}`}
              inverted={true}
              maxDepth={max}
              currentValue={entry.depth}
            >
              <td>{entry.priceStr}</td>
              <td>{entry.sizeStr}</td>
              <td>{entry.depthStr}</td>
            </TableRow>
          ))}
        </tbody>
      </table>
    </TableStyles>
  );
};
