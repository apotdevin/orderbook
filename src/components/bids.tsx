import { FC } from 'react';
import { OrderEntry } from 'src/hooks/useOrderbook';
import { TableRow, TableStyles } from 'src/components/table';
import { formatCurrency } from 'src/utils/format';
import { Typo } from './typo';

export const Bids: FC<{ entries: OrderEntry[]; max: number }> = ({
  entries,
  max,
}) => {
  const tableData = entries.map(b => ({
    ...b,
    depthStr: formatCurrency(b.depth),
    sizeStr: formatCurrency(b.size),
    priceStr: <Typo.bid>{formatCurrency(b.price)}</Typo.bid>,
    max,
  }));

  return (
    <TableStyles>
      <table>
        <tr>
          <td>TOTAL</td>
          <td>SIZE</td>
          <td>PRICE</td>
        </tr>
        {tableData.map(entry => (
          <TableRow
            key={entry.price}
            inverted={false}
            maxDepth={max}
            currentValue={entry.depth}
          >
            <td>{entry.depthStr}</td>
            <td>{entry.sizeStr}</td>
            <td>{entry.priceStr}</td>
          </TableRow>
        ))}
      </table>
    </TableStyles>
  );
};
