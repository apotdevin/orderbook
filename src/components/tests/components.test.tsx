import { render } from '@testing-library/react';
import { OrderMocks } from 'src/hooks/mocks/orders';
import { Asks } from '../asks';
import { Bids } from '../bids';
import { Header } from '../header';
import { TableRow } from '../table';

describe('Components', () => {
  describe('Asks Table', () => {
    test('matches snapshot', () => {
      const { asFragment } = render(
        <Asks entries={OrderMocks} max={5} isMobile={false} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
    test('matches mobile snapshot', () => {
      const { asFragment } = render(
        <Asks entries={OrderMocks} max={5} isMobile={true} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe('Bids Table', () => {
    test('matches snapshot', () => {
      const { asFragment } = render(
        <Bids entries={OrderMocks} max={5} isMobile={false} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
    test('matches mobile snapshot', () => {
      const { asFragment } = render(
        <Bids entries={OrderMocks} max={5} isMobile={true} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe('Header', () => {
    test('matches snapshot', () => {
      const { asFragment } = render(<Header />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe('TableRow', () => {
    test('matches snapshot', () => {
      const { asFragment } = render(
        <table>
          <tbody>
            <TableRow
              inverted={false}
              maxDepth={10}
              currentValue={2}
              left={false}
            >
              <td>1</td>
            </TableRow>
          </tbody>
        </table>
      );
      expect(asFragment()).toMatchSnapshot();
    });
    test('matches left snapshot', () => {
      const { asFragment } = render(
        <table>
          <tbody>
            <TableRow
              inverted={false}
              maxDepth={10}
              currentValue={2}
              left={true}
            >
              <td>1</td>
            </TableRow>
          </tbody>
        </table>
      );
      expect(asFragment()).toMatchSnapshot();
    });
    test('matches inverted snapshot', () => {
      const { asFragment } = render(
        <table>
          <tbody>
            <TableRow
              inverted={true}
              maxDepth={10}
              currentValue={2}
              left={false}
            >
              <td>1</td>
            </TableRow>
          </tbody>
        </table>
      );
      expect(asFragment()).toMatchSnapshot();
    });
    test('matches inverted left snapshot', () => {
      const { asFragment } = render(
        <table>
          <tbody>
            <TableRow
              inverted={true}
              maxDepth={10}
              currentValue={2}
              left={true}
            >
              <td>1</td>
            </TableRow>
          </tbody>
        </table>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
