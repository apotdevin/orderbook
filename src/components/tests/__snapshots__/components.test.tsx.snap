// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Asks Table matches mobile snapshot 1`] = `
<DocumentFragment>
  <div
    class="table__TableStyles-sc-1omlfwy-0 fMbACP"
  >
    <table>
      <thead>
        <tr>
          <td>
            PRICE
          </td>
          <td>
            SIZE
          </td>
          <td>
            TOTAL
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__ask-b9ovmy-1 bZSXIY"
            >
              30.00
            </div>
          </td>
          <td>
            7
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__ask-b9ovmy-1 bZSXIY"
            >
              10.00
            </div>
          </td>
          <td>
            1,000
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__ask-b9ovmy-1 bZSXIY"
            >
              2.00
            </div>
          </td>
          <td>
            120
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__ask-b9ovmy-1 bZSXIY"
            >
              1.00
            </div>
          </td>
          <td>
            100
          </td>
          <td>
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Components Asks Table matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="table__TableStyles-sc-1omlfwy-0 fMbACP"
  >
    <table>
      <thead>
        <tr>
          <td>
            PRICE
          </td>
          <td>
            SIZE
          </td>
          <td>
            TOTAL
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__ask-b9ovmy-1 bZSXIY"
            >
              1.00
            </div>
          </td>
          <td>
            100
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__ask-b9ovmy-1 bZSXIY"
            >
              2.00
            </div>
          </td>
          <td>
            120
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__ask-b9ovmy-1 bZSXIY"
            >
              10.00
            </div>
          </td>
          <td>
            1,000
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__ask-b9ovmy-1 bZSXIY"
            >
              30.00
            </div>
          </td>
          <td>
            7
          </td>
          <td>
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Components Bids Table matches mobile snapshot 1`] = `
<DocumentFragment>
  <div
    class="table__TableStyles-sc-1omlfwy-0 fMbACP"
  >
    <table>
      <thead>
        <tr>
          <td>
            PRICE
          </td>
          <td>
            SIZE
          </td>
          <td>
            TOTAL
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__bid-b9ovmy-0 dfCUYX"
            >
              1.00
            </div>
          </td>
          <td>
            100
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__bid-b9ovmy-0 dfCUYX"
            >
              2.00
            </div>
          </td>
          <td>
            120
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__bid-b9ovmy-0 dfCUYX"
            >
              10.00
            </div>
          </td>
          <td>
            1,000
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            <div
              class="typo__bid-b9ovmy-0 dfCUYX"
            >
              30.00
            </div>
          </td>
          <td>
            7
          </td>
          <td>
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Components Bids Table matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="table__TableStyles-sc-1omlfwy-0 fMbACP"
  >
    <table>
      <thead>
        <tr>
          <td>
            TOTAL
          </td>
          <td>
            SIZE
          </td>
          <td>
            PRICE
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            0
          </td>
          <td>
            100
          </td>
          <td>
            <div
              class="typo__bid-b9ovmy-0 dfCUYX"
            >
              1.00
            </div>
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            0
          </td>
          <td>
            120
          </td>
          <td>
            <div
              class="typo__bid-b9ovmy-0 dfCUYX"
            >
              2.00
            </div>
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            0
          </td>
          <td>
            1,000
          </td>
          <td>
            <div
              class="typo__bid-b9ovmy-0 dfCUYX"
            >
              10.00
            </div>
          </td>
        </tr>
        <tr
          class="table__TableRow-sc-1omlfwy-1"
        >
          <td>
            0
          </td>
          <td>
            7
          </td>
          <td>
            <div
              class="typo__bid-b9ovmy-0 dfCUYX"
            >
              30.00
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Components Header matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="header__Wrapper-w5cr8-0 bJCuER"
  >
    Orderbook Example
  </div>
</DocumentFragment>
`;

exports[`Components TableRow matches inverted left snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="table__TableRow-sc-1omlfwy-1"
      >
        <td>
          1
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Components TableRow matches inverted snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="table__TableRow-sc-1omlfwy-1"
      >
        <td>
          1
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Components TableRow matches left snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="table__TableRow-sc-1omlfwy-1"
      >
        <td>
          1
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Components TableRow matches snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="table__TableRow-sc-1omlfwy-1"
      >
        <td>
          1
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
